cmd/common/registry.go

```diff
@@ -310,17 +310,37 @@ func buildCommandFromGadget(
 			}
 
 			formatter := parser.GetTextColumnsFormatter()
-			if outputModeParams != "" {
-				valid, invalid := parser.VerifyColumnNames(strings.Split(outputModeParams, ","))
 
-				for _, c := range invalid {
-					log.Warnf("column %q not found", c)
+			requestedStandardColumns := outputModeParams == ""
```

Comment:
I tried to make it work for the tcpconnect gadget and I wasn't able. It turns out that the following logic is wrong:

https://github.com/inspektor-gadget/inspektor-gadget/blob/e2c674b962e71b08bb264a8df0eda389dbbd2306/cmd/common/registry.go#L486-L490

It'll change the value of `defaultOutputFormat` from  `columns=foo,bar,zas` to `columns` if the gadget declares that as default format. In other words, this `outputModeParams == ""` check is only valid for the `snapshot process` as it implements `OutputFormats` and returns `columns` ->  https://github.com/inspektor-gadget/inspektor-gadget/blob/b9eea85c9d1ed8ee8f7231c7a84a6950e00846e0/pkg/gadgets/snapshot/process/tracer/gadget.go#L90

I suppose this should be something like `outputModeParams == strings.Join(parser.GetDefaultColumns(), ",")`, but we need to be sure parser is not nil, etc...

Reply:
Ahh, I had it that way first, but when I tested it, the outputModeParams were empty, so I changed it 😆. Will fix it and do some more tests. Thanks!

Reply:
I've changed the default to be empty now - that way we can differentiate between the value being set explicitly by the user or not. Otherwise we would have also adjusted, if the user specified exactly the default columns.